internal/pkg/addon/package_test.go

```diff
@@ -23,12 +32,41 @@ func TestPackage(t *testing.T) {
 		bucket = "mockBucket"
 	)
 
+	lambdaZipHash := sha256.New()
+	indexZipHash := sha256.New()
+	indexFileHash := sha256.New()
+
+	// fs has the following structure:
+	//  .
+	//  ├─ lambda
+	//  │  ├─ index.js (contains lambda function)
+	//  ┴  └─ test.js (empty)
+	fs := afero.NewMemMapFs()
+	fs.Mkdir("/lambda", 0644)
+
+	f, _ := fs.Create("/lambda/index.js")
+	defer f.Close()
+	info, _ := f.Stat()
+	io.MultiWriter(lambdaZipHash, indexZipHash).Write([]byte("index.js " + info.Mode().String()))
+	io.MultiWriter(f, lambdaZipHash, indexZipHash, indexFileHash).Write([]byte(`exports.handler = function(event, context) {}`))
+
+	f2, _ := fs.Create("/lambda/test.js")
+	info, _ = f2.Stat()
+	lambdaZipHash.Write([]byte("test.js " + info.Mode().String()))
+
+	lambdaZipS3Path := artifactpath.AddonAsset(wlName, hex.EncodeToString(lambdaZipHash.Sum(nil)))
+	indexZipS3Path := artifactpath.AddonAsset(wlName, hex.EncodeToString(indexZipHash.Sum(nil)))
+	indexFileS3Path := artifactpath.AddonAsset(wlName, hex.EncodeToString(indexFileHash.Sum(nil)))
```

Comment:
For unit tests, I try to follow this rule as long as it doesn't make the test too difficult too difficult to follow:
>The lesson is clear: in test code, stick to straight-line code over clever logic, and consider tolerating some duplication when it makes the test more descriptive and meaningful.
> https://abseil.io/resources/swe-book/html/ch12.html#donapostrophet_put_logic_in_tests

I think the logic using the *ZipHash variables is fair, otherwise the reader of the test won't have any idea how the hashes are generated. But for the s3 paths, we can be slightly more explicit to ensure we're uploading to the correct location:
```go
wantedLambdaZipPath := fmt.Sprintf("manual/addons/mock-wl/assets/%s", hex.EncodeToString(lambdaZipHash.Sum(nil))) 
```